lib/python/rosie/ws.py

```diff
@@ -189,33 +224,114 @@ def _render(self, all_revs=0, data=None, filters=None, s=None):
             filters=filters,
             s=s,
             data=data)
+        )
+
+
+class HelloHandler(RosieDiscoService):
+
+    """Writes a 'Hello' message to the current logged-in user, else 'user'."""
+
+    HELLO = "Hello %s\n"
+
+    def get(self):
+        """Say Hello on success."""
+        if self.get_current_user():
```

Comment:
As far as I can see you haven't over-ridden `tornado.web.RequestHandler.get_current_user`. As (the method in the base class](https://www.tornadoweb.org/en/stable/_modules/tornado/web.html#RequestHandler.get_current_user) is a placeholder which always returns `None` this logic will never select the `if self.get_current_user():`. Is this desirable behaviour?

Reply:
(See comment under your next, associated, sub-review.)